hier_config/utils.py

```diff
@@ -29,5 +63,184 @@ def load_hier_config_tags(tags_file: str) -> tuple[TagRule, ...]:
         Tuple[TagRule, ...]: A tuple of validated TagRule objects.
 
     """
-    tags_data = yaml.safe_load(Path(tags_file).read_text(encoding="utf-8"))
+    tags_data = yaml.safe_load(_load_file_contents(file_path=tags_file))
     return TypeAdapter(tuple[TagRule, ...]).validate_python(tags_data)
+
+
+def hconfig_v2_os_v3_platform_mapper(os_name: str) -> Platform:
+    """Map a Hier Config v2 operating system name to a v3 Platform enumeration.
+
+    Args:
+        os_name (str): The name of the OS as defined in Hier Config v2.
+
+    Returns:
+        Platform: The corresponding Platform enumeration for Hier Config v3.
+
+    Example:
+        >>> hconfig_v2_os_v3_platform_mapper("CISCO_IOS")
+        <Platform.CISCO_IOS: 'ios'>
+
+    """
+    return hconfig_mapping.get(os_name, Platform.GENERIC)
+
+
+def hconfig_v3_platform_v2_os_mapper(platform: Platform) -> str:
+    """Map a Hier Config v3 Platform enumeration to a v2 operating system name.
+
+    Args:
+        platform (Platform): A Platform enumeration from Hier Config v3.
+
+    Returns:
+        str: The corresponding OS name for Hier Config v2.
+
+    Example:
+        >>> hconfig_v3_platform_v2_os_mapper(Platform.CISCO_IOS)
+        "ios"
+
+    """
+    for os_name, plat in hconfig_mapping.items():
+        if plat == platform:
+            return os_name
+
+    return "generic"
+
+
+def load_hconfig_v2_options(
+    v2_options: dict[str, Any], platform: Platform
+) -> HConfigDriverBase:
+    """Load Hier Config v2 options to v3 driver format.
+
+    Args:
+        v2_options (dict): The v2 options dictionary.
+        platform (Platform): The Hier Config v3 Platform enum for the target platform.
+
+    Returns:
+        HConfigDriverBase: A v3 driver instance with the migrated rules.
+
+    """
+    driver = get_hconfig_driver(platform)
+
+    # Map v2 options to v3 driver rules
+    if "negation" in v2_options:
+        driver.rules.negate_with.append(
+            NegationDefaultWithRule(
+                match_rules=(MatchRule(startswith=""),),  # Adjust match logic if needed
+                use=v2_options["negation"],
+            )
+        )
+
+    if "ordering" in v2_options:
+        for order in v2_options["ordering"]:
+            driver.rules.ordering.append(
+                OrderingRule(
+                    match_rules=(
+                        MatchRule(startswith=order["lineage"][0]["startswith"]),
+                    ),
+                    weight=order.get("order", 500),
```

Comment:
We should retrieve the `order` and use it here. Since 0 is now the default weight, we should subtract 500 from the weight.